src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -921,11 +921,65 @@ private bool ScanInteger()
             return start < TextWindow.Position;
         }
 
+        // Allows underscores in integers, except at beginning and end
+        private void ScanNumericLiteralSingleInteger(StringBuilder builder, ref bool underscoreInWrongPlace, ref bool usedUnderscore, bool isHex, bool isBinary)
+        {
+            if (TextWindow.PeekChar() == '_')
+            {
+                underscoreInWrongPlace = true;
+            }
+
+            char ch;
+            var lastCharWasUnderscore = false;
+            while (true)
```

Comment:
I think this test already does so - https://github.com/khyperia/roslyn/blob/NumericLiterals/src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalTests.cs#L2351 (not sure if this link will work for posterity, but LexicalTests.cs TestNumericBinaryLiteral())

As far as I understand it, it just invokes the lexer with the specified text, with an immediate EOF afterwards.